scripts/core/groupalgorithmmemory.yml

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension APIs for Querying Memory Properties of Group Algorithms"
+version: "1.6"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Group Algorithm Memory Properties Extension Name"
+version: "1.6"
+name: $X_DEVICE_GROUP_ALGORITHM_MEMORY_PROPERTIES_EXP_NAME
+value: '"$X_experimental_device_group_algorithm_memory_properties"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Group Algorithm Memory Properties version(s)"
+version: "1.6"
+name: $x_device_group_algorithm_memory_properties_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported group algorithms to be used with $xDeviceGetGroupAlgorithmMemoryPropertiesExp"
+version: "1.6"
+class: $xDevice
+name: $x_group_algorithm_type_exp_t
+etors:
+    - name: SORT
+      value: "0"
+      desc: "sorting algorithm"
+    - name: SCAN
+      desc: "scan algorithm"
+    - name: REDUCE
+      desc: "reduce algorithm"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported scopes in group algorithms to be used with $xDeviceGetGroupAlgorithmMemoryPropertiesExp"
+version: "1.6"
+class: $xDevice
+name: $x_group_algorithm_memory_scope_exp_t
+etors:
+    - name: SUBGROUP
+      value: "0"
+      desc: "memory scope limited to subgroup"
+    - name: WORKGROUP
+      desc: "memory scope limited to workgroup"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Device Group Algorithm Memory Properties"
+version: "1.6"
+class: $xDevice
+name: $x_device_group_algorithm_memory_exp_properties_t
+base: $x_base_properties_t
+members:
+    - type: $x_group_algorithm_type_exp_t
+      name: "algorithm"
+      desc: "[in] Type of group algorithm"
+    - type: $x_group_algorithm_memory_scope_exp_t
+      name: "memoryScope"
+      desc: "[in] Memory scope of group algorithm"
+    - type: size_t*
```

Comment:
i just realized, this needs to be scalar values, as the struct is already pointer.